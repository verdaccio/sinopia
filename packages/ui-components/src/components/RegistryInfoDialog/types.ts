import { ReactNode } from 'react';

export interface Props {
  children: ReactNode;
  open: boolean;
  title: string;
  onClose: () => void;
}
