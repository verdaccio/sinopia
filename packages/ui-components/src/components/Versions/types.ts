import { PackageMetaInterface } from '../../types/packageMeta';

export interface DetailContextProps {
  hasNotBeenFound: boolean;
  isLoading: boolean;
  packageMeta: PackageMetaInterface;
  packageName: string;
  packageVersion?: string;
  readMe: string;
}

export interface VersionPageConsumerProps {
  packageMeta: PackageMetaInterface;
  packageName: string;
  packageVersion?: string;
  readMe: string;
}
