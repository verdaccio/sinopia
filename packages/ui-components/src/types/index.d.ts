declare module '*.svg' {
  const content: string;
  export default content;
}

declare module '*.png' {
  const value: string;
  export = value;
}

declare module '*.jpg' {
  const value: string;
  export = value;
}

declare module '*.md' {
  const value: string;
  export = value;
}
