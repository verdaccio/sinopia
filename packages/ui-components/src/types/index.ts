import { TemplateUIOptions } from '@verdaccio/types';

declare global {
  interface Window {
    /**
     * __VERDACCIO_BASENAME_UI_OPTIONS is a global object available in the
     * browser.
     */
    __VERDACCIO_BASENAME_UI_OPTIONS: TemplateUIOptions;
  }
}
