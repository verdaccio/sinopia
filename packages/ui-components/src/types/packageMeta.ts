export type ModuleType = 'commonjs' | 'module';

export type Latest = {
  author?: string | Author;
  deprecated?: string;
  name: string;
  dist?: {
    fileCount: number;
    unpackedSize: number;
    tarball?: string;
  };
  engines?: {
    node?: string;
    npm?: string;
    pnpm?: string;
    yarn?: string;
  };
  license?: string | LicenseInterface;
  version: string;
  homepage?: string;
  bugs?: {
    url: string;
  };
  repository?: {
    type?: string;
    url?: string;
  };
  main?: string;
  type?: ModuleType;
  types?: string;
  description?: string;
  funding?: Funding;
  maintainers?: Developer[];
  contributors?: Developer[];
  keywords?: string | string[];
};

export interface PackageMetaInterface {
  versions?: Versions;
  'dist-tags'?: DistTags;
  time?: Time;
  latest: Latest;
  _uplinks?: Record<string, { fetched: number }>;
}

export interface Developer {
  name: string;
  email?: string;
  url?: string;
  avatar?: string;
}

interface Funding {
  type?: string;
  url?: string;
}

export interface LicenseInterface {
  type?: string;
  url?: string;
}

export interface DistTags {
  [key: string]: string;
}

export interface Time {
  [key: string]: string;
}

export interface Versions {
  [key: string]: Version;
}

export interface Version {
  name: string;
  version: string;
  author?: string | Author;
  description?: string;
  license?: string | LicenseInterface;
  main?: string;
  keywords?: string[];
  deprecated?: string;
}

export interface Author {
  name?: string;
  email?: string;
  url?: string;
  avatar?: string;
}

export interface PackageDependencies {
  [key: string]: string;
}
