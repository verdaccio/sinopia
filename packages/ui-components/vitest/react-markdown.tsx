import React from 'react';
import { vi } from 'vitest';

// Mock react-markdown globally
vi.mock('react-markdown', () => ({
  default: ({ children }: { children: React.ReactNode }) => <>{children}</>,
}));
