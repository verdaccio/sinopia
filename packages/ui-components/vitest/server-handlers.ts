import fs from 'fs';
import { HttpResponse, http } from 'msw';
import path from 'path';

export const handlers = [
  http.get('http://localhost:9000/-/verdaccio/data/packages', () => {
    return HttpResponse.json(Array(400).fill(require('./api/home-packages.json')));
  }),

  http.get('http://localhost:9000/-/verdaccio/data/sidebar/storybook', ({ params }) => {
    const { v } = params;
    if (v) {
      return HttpResponse.json(require('./api/storybook-v.json'));
    } else {
      return HttpResponse.json(require('./api/storybook-sidebar.json'));
    }
  }),

  http.get('http://localhost:9000/-/verdaccio/data/search/*', () => {
    return HttpResponse.json(require('./api/search-verdaccio.json'));
  }),

  http.get('http://localhost:9000/-/verdaccio/data/package/readme/storybook', () => {
    return HttpResponse.text(require('./api/storybook-readme')());
  }),

  http.get('http://localhost:9000/-/verdaccio/data/sidebar/jquery', () => {
    return HttpResponse.json(require('./api/jquery-sidebar.json'));
  }),
  http.get('http://localhost:9000/-/verdaccio/data/sidebar/JSONStream', () => {
    return new HttpResponse('unauthorized', { status: 401 });
  }),
  http.get('http://localhost:9000/-/verdaccio/data/sidebar/semver', () => {
    return new HttpResponse('internal server error', { status: 500 });
  }),
  http.get('http://localhost:9000/-/verdaccio/data/sidebar/kleur', () => {
    return new HttpResponse('not found', { status: 404 });
  }),
  http.get('http://localhost:9000/-/verdaccio/data/sidebar/glob', () => {
    return HttpResponse.json(require('./api/glob-sidebar.json'));
  }),
  http.get('http://localhost:9000/-/verdaccio/data/package/readme/glob', () => {
    return HttpResponse.text('foo glob');
  }),
  http.get('http://localhost:9000/-/verdaccio/data/sidebar/got', () => {
    return HttpResponse.json(require('./api/got-sidebar.json'));
  }),
  http.get('http://localhost:9000/-/verdaccio/data/package/readme/got', () => {
    return HttpResponse.text('foo got');
  }),
  http.get('http://localhost:9000/-/verdaccio/data/package/readme/jquery', () => {
    return HttpResponse.text(require('./api/jquery-readme')());
  }),
  http.get('http://localhost:9000/verdaccio/-/verdaccio-1.0.0.tgz', () => {
    const fileName = path.resolve(__dirname, './api/verdaccio-1.0.0.tgz');
    const fileContent = fs.readFileSync(fileName);
    return new HttpResponse(fileContent, {
      status: 200,
      headers: { 'Content-Type': 'application/octet-stream' },
    });
  }),

  http.post<{ username: string; password: string }, { token: string; username: string }>(
    'http://localhost:9000/-/verdaccio/sec/login',
    async ({ request }) => {
      const body = await request.json();
      if (body.username === 'fail') {
        return new HttpResponse('unauthorized', { status: 401 });
      }

      return HttpResponse.json({ username: body.username, token: 'valid token' });
    }
  ),
];
