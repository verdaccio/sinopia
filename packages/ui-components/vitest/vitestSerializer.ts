import { SnapshotSerializer } from 'vitest';

const serializer: SnapshotSerializer = {
  serialize(val, config, indentation, depth, refs, printer) {
    for (const className of [...val.classList]) {
      if (className.startsWith('css-')) {
        const hashEnd = className.indexOf('-', 4);
        if (hashEnd >= 0) {
          val.classList.replace(className, 'makeStyles' + className.substring(hashEnd));
        } else {
          val.classList.remove(className);
        }
      }
    }
    if (!val.classList.length) {
      val.removeAttribute('class');
    }
    return printer(val, config, indentation, depth, refs);
  },
  test(val) {
    return val instanceof Element && [...val.classList].some((name) => name.startsWith('css-'));
  },
};

/**
 * An object snapshot serializer that removes dynamic CSS classes (generated by makeStyle) from DOM elements.
 *
 * `css-hash-suffix` is transformed into `makeStyles-suffix`
 * `css-hash` is removed
 */
export default serializer;
