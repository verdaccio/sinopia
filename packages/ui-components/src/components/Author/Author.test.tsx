import React from 'react';
import { vi } from 'vitest';

import { cleanup, render, screen } from '../../test/test-react-testing-library';
import { PackageMetaInterface } from '../../types/packageMeta';
import Authors from './Author';

const withAuthorComponent = (packageMeta: PackageMetaInterface): JSX.Element => (
  <Authors packageMeta={packageMeta} />
);

describe('<Author /> component', () => {
  beforeEach(() => {
    vi.resetAllMocks();
    cleanup();
  });

  test('should render the component in default state', () => {
    const packageMeta = {
      latest: {
        name: 'verdaccio',
        version: '4.0.0',
        author: {
          name: 'verdaccio user',
          email: 'verdaccio.user@verdaccio.org',
          url: '',
          avatar: 'https://www.gravatar.com/avatar/000000',
        },
        dist: { fileCount: 0, unpackedSize: 0 },
      },
      _uplinks: {},
    };

    const wrapper = render(withAuthorComponent(packageMeta));
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the component when there is no author information available', () => {
    const packageMeta = {
      latest: {
        name: 'verdaccio',
        version: '4.0.0',
        dist: { fileCount: 0, unpackedSize: 0 },
      },
      _uplinks: {},
    };

    const wrapper = render(withAuthorComponent(packageMeta));
    expect(wrapper.queryAllByText('verdaccio')).toHaveLength(0);
  });

  test('should render the component when there is no author email', () => {
    const packageMeta = {
      latest: {
        name: 'verdaccio',
        version: '4.0.0',
        author: {
          name: 'verdaccio user',
          url: '',
          avatar: 'https://www.gravatar.com/avatar/000000',
        },
        dist: { fileCount: 0, unpackedSize: 0 },
      },
      _uplinks: {},
    };

    const wrapper = render(withAuthorComponent(packageMeta));
    expect(wrapper).toMatchSnapshot();
  });

  test('should not render if data is missing', () => {
    // @ts-ignore - testing with missing data
    render(withAuthorComponent(undefined));
    expect(screen.queryByText('sidebar.author.title')).toBeNull();
  });
});
