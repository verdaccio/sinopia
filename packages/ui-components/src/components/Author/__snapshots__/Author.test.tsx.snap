// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Author /> component > should render the component in default state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root MuiList-padding MuiList-subheader makeStyles-MuiList-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 e4ifo371 makeStyles-MuiTypography-root-StyledText"
        >
          sidebar.author.title
        </h6>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding e4ifo370 makeStyles-MuiListItem-root-AuthorListItem"
        >
          <a
            class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineHover makeStyles-MuiTypography-root-MuiLink-root"
            data-mui-internal-clone-element="true"
            data-testid="verdaccio user"
            href="mailto:verdaccio.user@verdaccio.org?subject=verdaccio v4.0.0"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular makeStyles-MuiAvatar-root"
            >
              <img
                alt="verdaccio user"
                class="MuiAvatar-img makeStyles-MuiAvatar-img"
                src="https://www.gravatar.com/avatar/000000"
              />
            </div>
          </a>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 makeStyles-MuiTypography-root"
          >
            verdaccio user
          </h6>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root MuiList-padding MuiList-subheader makeStyles-MuiList-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 e4ifo371 makeStyles-MuiTypography-root-StyledText"
      >
        sidebar.author.title
      </h6>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding e4ifo370 makeStyles-MuiListItem-root-AuthorListItem"
      >
        <a
          class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineHover makeStyles-MuiTypography-root-MuiLink-root"
          data-mui-internal-clone-element="true"
          data-testid="verdaccio user"
          href="mailto:verdaccio.user@verdaccio.org?subject=verdaccio v4.0.0"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular makeStyles-MuiAvatar-root"
          >
            <img
              alt="verdaccio user"
              class="MuiAvatar-img makeStyles-MuiAvatar-img"
              src="https://www.gravatar.com/avatar/000000"
            />
          </div>
        </a>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 makeStyles-MuiTypography-root"
        >
          verdaccio user
        </h6>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Author /> component > should render the component when there is no author email 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root MuiList-padding MuiList-subheader makeStyles-MuiList-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 e4ifo371 makeStyles-MuiTypography-root-StyledText"
        >
          sidebar.author.title
        </h6>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding e4ifo370 makeStyles-MuiListItem-root-AuthorListItem"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular makeStyles-MuiAvatar-root"
            data-mui-internal-clone-element="true"
            data-testid="verdaccio user"
          >
            <img
              alt="verdaccio user"
              class="MuiAvatar-img makeStyles-MuiAvatar-img"
              src="https://www.gravatar.com/avatar/000000"
            />
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 makeStyles-MuiTypography-root"
          >
            verdaccio user
          </h6>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root MuiList-padding MuiList-subheader makeStyles-MuiList-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 e4ifo371 makeStyles-MuiTypography-root-StyledText"
      >
        sidebar.author.title
      </h6>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding e4ifo370 makeStyles-MuiListItem-root-AuthorListItem"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular makeStyles-MuiAvatar-root"
          data-mui-internal-clone-element="true"
          data-testid="verdaccio user"
        >
          <img
            alt="verdaccio user"
            class="MuiAvatar-img makeStyles-MuiAvatar-img"
            src="https://www.gravatar.com/avatar/000000"
          />
        </div>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 makeStyles-MuiTypography-root"
        >
          verdaccio user
        </h6>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
